gemma/gemini_client.cc

```diff
@@ -0,0 +1,382 @@
+    if (use_https_) {
+      ssl_client_ = std::make_unique<httplib::SSLClient>(host, port);
+      ssl_client_->set_read_timeout(60, 0);
+      ssl_client_->set_write_timeout(60, 0);
+      ssl_client_->enable_server_certificate_verification(false);
+    } else {
+      client_ = std::make_unique<httplib::Client>(host, port);
+      client_->set_read_timeout(60, 0);
+      client_->set_write_timeout(60, 0);
+    }
+  }
+
+  // Unified request processing for both public and local APIs
+  json ProcessRequest(const json& request, bool stream = true) {
+    bool is_public_api = !api_key_.empty();
+    
+    std::string endpoint;
+    if (is_public_api) {
+      endpoint = stream ? "/v1beta/models/gemini-2.0-flash:streamGenerateContent?alt=sse" 
+                        : "/v1beta/models/gemini-2.0-flash:generateContent";
+    } else {
+      endpoint = stream ? "/v1beta/models/gemma-2b:streamGenerateContent" 
+                        : "/v1beta/models/gemma-2b:generateContent";
+    }
+    
+    // Only show verbose output in non-interactive mode
+    if (!interactive_mode_) {
+      std::cout << "\n" << BOLD << BLUE << "📤 POST " << endpoint << RESET << std::endl;
+      std::cout << "Request: " << request.dump(2) << std::endl;
+    }
+    
+    if (stream) {
+      return ProcessStreamingRequest(request, endpoint);
+    } else {
+      return ProcessNonStreamingRequest(request, endpoint);
+    }
+  }
+
+  void TestGenerateContent(const std::string& prompt, bool stream = true) {
+    json request = CreateGeminiRequest(prompt);
+    json response = ProcessRequest(request, stream);
+    
+    if (response.contains("error")) {
+      std::cerr << RED << "❌ Error: " << response["error"]["message"] << RESET << std::endl;
+    }
+  }
+
+  void TestListModels() {
+    std::cout << "\n" << BOLD << BLUE << "📤 GET /v1beta/models" << RESET << std::endl;
+    
+    httplib::Headers headers;
+    if (!api_key_.empty()) {
+      headers.emplace("X-goog-api-key", api_key_);
+    }
+    auto res = use_https_ ? ssl_client_->Get("/v1beta/models", headers) : client_->Get("/v1beta/models", headers);
+    
+    if (res && res->status == 200) {
+      json response = json::parse(res->body);
+      std::cout << GREEN << "✅ Available models:" << RESET << std::endl;
+      std::cout << response.dump(2) << std::endl;
+    } else {
+      std::cerr << RED << "❌ Request failed" << RESET << std::endl;
+    }
+  }
+
+  void InteractiveChat() {
+    std::cout << "\n" << BOLD << CYAN << "💬 Interactive Chat Mode (with session)" << RESET << std::endl;
+    std::cout << "Type 'quit', 'exit', or 'q' to end.\n" << std::endl;
+    
+    interactive_mode_ = true;
+    json messages;
+    
+    while (true) {
+      std::cout << BOLD << BLUE << "You: " << RESET;
+      std::string input;
+      std::getline(std::cin, input);
+      
+      if (input == "quit" || input == "exit" || input == "q") {
```

Comment:
Let's also support %q as used in the `:gemma` repl?

Reply:
Changed to only accept ':gemma %q'.